### DISCUSSION TITLE
Feature request: AVIF/JXL image formats support and title as caption

### DISCUSSION THREAD

**Discussion by LITUATUI** [hbstack/theme-cards#225] [Ideas]
Hi.

I'm almost done migrating my blog to this Hugo theme. It has almost everything that I need.

Only two things missing.

It would be nice to make [AVIF](https://caniuse.com/avif) and [JXL](https://caniuse.com/jpegxl) supported image formats. I decided to replace all my JPG and PNG files with AVIF and got a bunch of errors (unknown format):

`ERROR render of "term" failed: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog@v0.25.2/layouts/_default/term.html:10:9": execute of template failed: template: _default/term.html:10:9: executing "main" at <partialCached "hb/modules/blog/term/profile" . .>: error calling partialCached: execute of template failed: template: partials/hb/modules/blog/term/profile.html:3:5: executing "partials/hb/modules/blog/term/profile.html" at <partial "hb/modules/blog/author/image" (dict "Page" . "Placeholder" false "ClassName" "hb-blog-term-img img-fluid rounded-circle mb-3")>: error calling partial: execute of template failed: template: partials/hb/modules/blog/author/image.html:5:5: executing "partials/hb/modules/blog/author/image.html" at <partial "images/image" (dict "Filename" (index $page.Params.images 0) "Page" $page "Alt" $page.Title "ClassName" $className)>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/images@v0.10.0/layouts/partials/images/image.html:162:15": execute of template failed: template: partials/images/image.html:162:15: executing "partials/images/image.html" at <imageConfig $filepath>: error calling imageConfig: image: unknown format
ERROR render of "taxonomy" failed: execute of template failed: template: authors/terms.html:9:26: executing "main" at <partial "hb/modules/blog/author/image" (dict "Page" . "Placeholder" false "ClassName" "hb-blog-term-img img-fluid rounded-circle mb-3")>: error calling partial: execute of template failed: template: partials/hb/modules/blog/author/image.html:5:5: executing "partials/hb/modules/blog/author/image.html" at <partial "images/image" (dict "Filename" (index $page.Params.images 0) "Page" $page "Alt" $page.Title "ClassName" $className)>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/images@v0.10.0/layouts/partials/images/image.html:162:15": execute of template failed: template: partials/images/image.html:162:15: executing "partials/images/image.html" at <imageConfig $filepath>: error calling imageConfig: image: unknown format
ERROR render of "taxonomy" failed: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/theme-cards@v0.1.1/layouts/_default/terms.html:26:5": execute of template failed: template: _default/terms.html:26:5: executing "main" at <partial "hb/modules/blog/sidebar" .>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog@v0.25.2/layouts/partials/hb/modules/blog/sidebar.html:2:5": execute of template failed: template: partials/hb/modules/blog/sidebar.html:2:5: executing "partials/hb/modules/blog/sidebar.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "hb-blog-sidebar" "Page" .)>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/functions/render-hooks.html:18:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:18:8: executing "partials/hugopress/functions/render-hooks.html" at <partialCached .partial $ctx>: error calling partialCached: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog/modules/sidebar/posts@v0.7.0/layouts/partials/hugopress/modules/hb-blog-sidebar-posts/hooks/hb-blog-sidebar.html:61:13": execute of template failed: template: partials/hugopress/modules/hb-blog-sidebar-posts/hooks/hb-blog-sidebar.html:61:13: executing "partials/hugopress/modules/hb-blog-sidebar-posts/hooks/hb-blog-sidebar.html" at <partial "hb/modules/blog-sidebar-posts/index" .>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog/modules/sidebar/posts@v0.7.0/layouts/partials/hb/modules/blog-sidebar-posts/index.html:5:5": execute of template failed: template: partials/hb/modules/blog-sidebar-posts/index.html:5:5: executing "partials/hb/modules/blog-sidebar-posts/index.html" at <partial "hb/modules/blog-sidebar-posts/slide" .>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog/modules/sidebar/posts@v0.7.0/layouts/partials/hb/modules/blog-sidebar-posts/slide.html:6:11": execute of template failed: template: partials/hb/modules/blog-sidebar-posts/slide.html:6:11: executing "partials/hb/modules/blog-sidebar-posts/slide.html" at <partial "hb/modules/blog/post/card" $ctx>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog@v0.25.2/layouts/partials/hb/modules/blog/post/card.html:20:7": execute of template failed: template: partials/hb/modules/blog/post/card.html:20:7: executing "partials/hb/modules/blog/post/card.html" at <partial "hb/modules/blog/post/card-img" $page>: error calling partial: execute of template failed: template: partials/hb/modules/blog/post/card-img.html:21:7: executing "partials/hb/modules/blog/post/card-img.html" at <partial "images/image" (dict "Page" $page "Alt" $page.Title "Filename" $img "ClassName" "hb-blog-post-card-img" "Original" false)>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/images@v0.10.0/layouts/partials/images/image.html:162:15": execute of template failed: template: partials/images/image.html:162:15: executing "partials/images/image.html" at <imageConfig $filepath>: error calling imageConfig: image: unknown format
Built in 2701 ms
Error: error building site: render: failed to render pages: render of "term" failed: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/blog@v0.25.2/layouts/_default/term.html:10:9": execute of template failed: template: _default/term.html:10:9: executing "main" at <partialCached "hb/modules/blog/term/profile" . .>: error calling partialCached: execute of template failed: template: partials/hb/modules/blog/term/profile.html:3:5: executing "partials/hb/modules/blog/term/profile.html" at <partial "hb/modules/blog/author/image" (dict "Page" . "Placeholder" false "ClassName" "hb-blog-term-img img-fluid rounded-circle mb-3")>: error calling partial: execute of template failed: template: partials/hb/modules/blog/author/image.html:5:5: executing "partials/hb/modules/blog/author/image.html" at <partial "images/image" (dict "Filename" (index $page.Params.images 0) "Page" $page "Alt" $page.Title "ClassName" $className)>: error calling partial: "/home/pedro/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/images@v0.10.0/layouts/partials/images/image.html:162:15": execute of template failed: template: partials/images/image.html:162:15: executing "partials/images/image.html" at <imageConfig $filepath>: error calling imageConfig: image: unknown format
`

Another thing missing, is to be able to use image titles as captions, like you have in your [old theme](https://hbs.razonyang.com/v1/en/docs/image-processing/#caption).

By the way, I have my images in the static folder instead of assets, because I don't want Hugo to try to convert them to WebP.

Thanks for your awesome theme, it's by far the best I tried yet!

**Comment by razonyang** [comment]
Seems buggy, I'm checking this.

**Comment by razonyang** [comment]
Please upgrade `github.com/hugomods/images` to `v0.10.1` and check if it works.

> Another thing missing, is to be able to use image titles as captions, like you have in your [old theme](https://hbs.razonyang.com/v1/en/docs/image-processing/#caption).

Should be implemented already.

> By the way, I have my images in the static folder instead of assets, because I don't want Hugo to try to convert them to WebP.

Yes, Hugo doesn't support using avif as image resources, put them into `assets` will produce errors.

**Comment by razonyang** [comment]
Btw, the [new docs site for images](https://images.hugomods.com/) is live now, usages are similar to the old theme.

**Comment by LITUATUI** [comment]
It works now, you're the best. Thanks!
